common/src/main/java/com/wynntils/gui/screens/ScreenWrapper.java

```diff
@@ -0,0 +1,313 @@
+/*
+ * Copyright © Wynntils 2022.
+ * This file is released under AGPLv3. See LICENSE for full license details.
+ */
+package com.wynntils.gui.screens;
+
+import com.mojang.blaze3d.vertex.PoseStack;
+import com.wynntils.core.WynntilsMod;
+import com.wynntils.mc.utils.McUtils;
+import java.nio.file.Path;
+import java.util.List;
+import java.util.Optional;
+import java.util.function.BiConsumer;
+import net.minecraft.client.Minecraft;
+import net.minecraft.client.gui.components.Widget;
+import net.minecraft.client.gui.components.events.GuiEventListener;
+import net.minecraft.client.gui.narration.NarratableEntry;
+import net.minecraft.client.gui.screens.Screen;
+import net.minecraft.network.chat.Component;
+import net.minecraft.network.chat.Style;
+import net.minecraft.network.chat.TextComponent;
+import net.minecraft.util.FormattedCharSequence;
+import net.minecraft.world.inventory.tooltip.TooltipComponent;
+import net.minecraft.world.item.ItemStack;
+
+public class ScreenWrapper extends Screen {
+    private final Screen delegate;
```

Comment:
How well does this work with instanceof checks? Not really well.. Is there any place we use instanceof for our custom screens? (I know we do for MC screens)

Reply:
Perhaps, make some util which checks instanceof, but if the current instance is this wrapper, it can handle it as well.

Reply:
We do that in once place, in the CharacterSelection screen. See the patch for that file. I can turn that into a util method, maybe it's a good idea.